common/memorypool.h

```diff
@@ -26,6 +26,14 @@
 #include "common/scummsys.h"
 #include "common/array.h"
 
+#ifndef NDEBUG
```

Comment:
We do not want to enable those checks in release builds, and currently we do not define `NDEBUG` in our release builds. I don't see any good reason for it, so I created PR #2486 to add it. But if it turns out there was a good reason not to define `NDEBUG`, this line will need to be changed to check `RELEASE_BUILD` instead.

Reply:
It looks like adding NDEBUG for release builds might not happen anytime soon. Can you please replace `#ifndef NDEBUG` with `#ifndef RELEASE_BUILD` and then I think we can merge this?